fs/base.py

```diff
@@ -1192,7 +1195,7 @@ def opendir(
         """
         from .subfs import SubFS
 
-        _factory = factory or SubFS
+        _factory = factory or self.subfs_class or SubFS
```

Comment:
Why not just define `subfs_class = SubFS` directly when declaring the class instead of letting it be `None` ?

Reply:
As I said above, it caused circular import dependencies because it means importing SubFS at the top of the module, but SubFS depends on FS.
I'm happy to change it if someone can suggest how it can work. Or y'all can reject it and I'll see if I can work it out myself.

Reply:
No it's fine, I had forgotten about that indeed.